### DISCUSSION TITLE
How to use LuaCov with mini.test?

### DISCUSSION THREAD

**Discussion by MunifTanjim** [nvim-mini/mini.nvim#589] [Q&A]
How to use [LuaCov](https://github.com/lunarmodules/luacov) with `mini.test`?

I tried `require`-ing LuaCov by doing `pcall(require, "luacov")` at the top of the test file. But it doesn't generate `luacov.stats.out` file after test is run.

**Comment by echasnovski** [comment]
Not quite sure, as I don't use it at all. It might be even impossible for how 'mini.test' recommends to do tests (each one in a separate new child process), so not sure.

**Reply by MunifTanjim** [reply]
Hmm... couldn't make it work. I guess only `plenary.nvim` supports it right now. 😅

**Reply by echasnovski** [reply]
Sorry to hear that.

I honestly (after reading several times through README and docs) don't understand how this can even work in a normal setting. Here, in 'mini.test', it is more like calling code in separate child processes, so even more hard. Maybe adding `require('luacov')` in 'minimal_init.lua' of child process can make it work?

**Reply by MunifTanjim** [reply]
Does `mini.test` spawn child process by default?

I tried with a very simple test:

```lua
pcall(require, "luacov")

describe("bar", function()
  it("init", function()
    local Bar = require("nougat.bar")
    local bar = Bar("statusline")
    MiniTest.expect.equality(bar, bar)
  end)
end)
```

I also tried doing `pcall(require, "luacov")` in the `init.lua` file, and running the test like this:

```
  nvim --headless --noplugin -u tests/init.lua \
    -c "lua MiniTest.run({ execute = { reporter = MiniTest.gen_reporter.stdout({ group_depth = 9 }) } })"
```

**Reply by echasnovski** [reply]
> Does `mini.test` spawn child process by default?
No, it doesn't.

------

Not sure, maybe it has something to do with the fact that 'mini.test' doesn't actually run tests when they are sourced. It first collects them in a special table and then executes its content.

**Reply by powerman** [reply]
Any progress on this? It will be sad to go back to `plenary.test_harness` just because of this issue. :disappointed:

**Reply by echasnovski** [reply]
> Any progress on this? It will be sad to go back to `plenary.test_harness` just because of this issue. 😞

There is no work done or planned for LuaCov compatibility with 'mini.test', I am afraid. Mostly because I can not reasonably imagine how this would work with the suggested way of running tests (each in a separate child process).

**Reply by powerman** [reply]
I've no idea about luacov, but I did similar things with coverage in Go - it was quite easy to merge coverage reports generated from different processes into common report. Maybe something like that will solve this issue for `mini.test` child processes too.

**Reply by echasnovski** [reply]
> ... it was quite easy to merge coverage reports generated from different processes into common report. Maybe something like that will solve this issue for `mini.test` child processes too.

Reserving so much effort in 'mini.test' itself for it to work with third party tool is definitely out of scope. The module is already pretty big as is.
